src/test/memory/test_mmap.rs

```diff
@@ -0,0 +1,354 @@
+/*
+ * The Shadow Simulator
+ * See LICENSE for licensing information
+ */
+
+use test_utils::set;
+use test_utils::TestEnvironment as TestEnv;
+
+const MAPLEN: usize = 16;
+
+fn validate_shadow_access(buf: &[u8]) -> Result<(), String> {
+    let template = b"test_mmapXXXXXX";
+
+    /* Get a file that we can mmap and write into. */
+    let (temp_fd, path) = nix::unistd::mkstemp(template.as_ref()).map_err(|e| e.to_string())?;
+    nix::errno::Errno::result(temp_fd).map_err(|e| e.to_string())?;
+
+    /* Make sure there is enough space to write after the mmap. */
+    {
```

Comment:
I don't think we need to call `posix_fallocate` here, since we're writing to it via `write` rather than `mmap`. The `write` will automatically add the necessary storage to the file.